airbyte-server/src/main/java/io/airbyte/server/handlers/OperationsHandler.java

```diff
@@ -144,6 +175,40 @@ public OperationRead getOperation(OperationIdRequestBody operationIdRequestBody)
     return buildOperationRead(operationIdRequestBody.getOperationId());
   }
 
+  public void deleteOperationsForConnection(ConnectionIdRequestBody connectionIdRequestBody)
+      throws JsonValidationException, ConfigNotFoundException, IOException {
+    final StandardSync standardSync = configRepository.getStandardSync(connectionIdRequestBody.getConnectionId());
+    deleteOperationsForConnection(standardSync, standardSync.getOperationIds());
+  }
+
+  public void deleteOperationsForConnection(UUID connectionId, List<UUID> deleteOperationIds)
+      throws JsonValidationException, ConfigNotFoundException, IOException {
+    final StandardSync standardSync = configRepository.getStandardSync(connectionId);
+    deleteOperationsForConnection(standardSync, deleteOperationIds);
+  }
+
+  public void deleteOperationsForConnection(StandardSync standardSync, List<UUID> deleteOperationIds)
+      throws JsonValidationException, ConfigNotFoundException, IOException {
+    final List<StandardSync> allStandardSyncs = configRepository.listStandardSyncs();
+    final List<UUID> operationIds = standardSync.getOperationIds();
+    for (UUID operationId : deleteOperationIds) {
+      operationIds.remove(operationId);
+      boolean sharedOperation = false;
+      for (StandardSync sync : allStandardSyncs) {
+        // Check if other connections are using the same operation
+        if (sync.getConnectionId() != standardSync.getConnectionId() && sync.getOperationIds().contains(operationId)) {
+          sharedOperation = true;
+          break;
+        }
+      }
+      if (!sharedOperation) {
```

Comment:
Depending on where the UI goes in the future we may have to change the behavior for deleting connections to not delete even unshared operations, but this should be fine for now.